### PR TITLE
Added new StackOverflow Endpoints

### PR DESCRIPTION
**Pull request by soumitradev**
Added
- featuredquestions: https://api.stackexchange.com/docs/featured-questions
- unanswered: https://api.stackexchange.com/docs/unanswered-questions
- noanswers: https://api.stackexchange.com/docs/no-answer-questions
- moderators: https://api.stackexchange.com/docs/moderators

I actually define a new struct in moderators.jl. Do I need to include this in StackOverflow.jl like the questions struct?

Also, the terms `unanswered` and `noanswers` seem a bit ambiguous to me, so I called the unanswered.jl function `getopenquestions()` since the question has answers, but remains open. Can someone help with this wording?

### CODE REVIEW

File: src/featuredquestions.jl [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    end
+
+    json = StackOverflow.convert_HTTP_Response_To_JSON(r)
+
+    questionholder = Vector()
+    for (k, v) in json
+        if occursin("items", k)
+            for item in v
+
+                link                = Vector()
```
Comment (logankilpatrick) [inline src/featuredquestions.jl]: I don't think these need to be vectors.

File: src/moderators.jl [hunk 2]
```diff
@@ -0,0 +1,169 @@
+using StackOverflow
+
+mutable struct Person
+    view_count::Vector
```
Comment (logankilpatrick) [inline src/moderators.jl]: Same here, I don't think we need vectors.

File: src/noanswers.jl [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    end
+
+    json = StackOverflow.convert_HTTP_Response_To_JSON(r)
+
+    questionholder = Vector()
+    for (k, v) in json
+        if occursin("items", k)
+            for item in v
+
+                link                = Vector()
```
Comment (logankilpatrick) [inline src/noanswers.jl]: Same here

**Review by logankilpatrick — CHANGES_REQUESTED** [review]
Looks good for the most part, but those vectors should be changed.

**Comment by soumitradev** [comment]
Sometimes, numbers like `view_count` may not be provided by the API reply, and 0 is not a valid value since it says that the user has never been viewed, which we don't know for sure. What do I do in such a case? @logankilpatrick

**Comment by soumitradev** [comment]
Just made the requested changes! For now, I have made all the Ints 0 by default.

File: src/featuredquestions.jl (hunk 1 above)
Comment (logankilpatrick) [reply src/featuredquestions.jl]: @soumitradev this should still change

File: src/noanswers.jl (hunk 3 above)
Comment (logankilpatrick) [reply src/noanswers.jl]: still need to fix this

File: src/unanswered.jl [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    end
+
+    json = StackOverflow.convert_HTTP_Response_To_JSON(r)
+
+    questionholder = Vector()
+    for (k, v) in json
+        if occursin("items", k)
+            for item in v
+
+                link                = Vector()
```
Comment (logankilpatrick) [inline src/unanswered.jl]: still need to fix this

**Comment by soumitradev** [comment]
Made the requested changes.

**Review by logankilpatrick — APPROVED** [review]
Nice work!

**Pull request merged**